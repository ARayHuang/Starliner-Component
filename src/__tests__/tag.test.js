import React from 'react';
import { shallow, mount } from 'enzyme';
import Tag from '../components/tag';
import { CheckIcon } from '@chakra-ui/icons'

describe('Tag', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-Tag-className',	
		};

		const wrapper = shallow(
			<Tag {...props}>
				<div>mock-children</div>
			</Tag>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-Tag-className',
			variant: 'outline',
			size: 'lg',
			showCloseBtn: true,
			text: 'mock-text',
			icon: CheckIcon,
			onClickBtn: () => {}		
		};
		
		const wrapper = mount(
			<Tag {...props}>
				mock-text
			</Tag>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().variant).toEqual(props.variant);
		expect(wrapper.props().size).toEqual(props.size);
		expect(wrapper.props().showCloseBtn).toEqual(props.showCloseBtn);
		expect(wrapper.props().icon).toEqual(props.icon);
		expect(wrapper.props().children).toEqual(props.text);
	})
})
