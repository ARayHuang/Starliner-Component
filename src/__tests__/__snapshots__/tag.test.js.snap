// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render correctly 1`] = `
<Fragment>
  <Tag
    className="mock-Tag-className"
    size="sm"
    sx={Object {}}
    variant="solid"
  >
    <TagLabel />
  </Tag>
</Fragment>
`;
